course4/public/src/drawing.js

```diff
@@ -47,6 +47,26 @@ Circle.prototype.drawFrame = function () {
   ctx.fill();
 };
 
+// Arc constructor
+function Arc(x, y, r, startAngle, endAngle, counterClockwise, lineWidth = 10, strokeStyle = 'blue') {
+  Shape.call(this, x, y);
+  this.r = r;
+  this.startAngle = startAngle * (Math.PI/180);
+  this.endAngle = endAngle * (Math.PI/180);
+  this.counterClockWise = counterClockwise;
+};
+// Arc extends Shape
+Arc.prototype = Object.create(Shape.prototype);
+//extend the drawFrame
+Arc.prototype.drawFrame = function() {
+  ctx.fillStyle = this.fill;
+  ctx.beginPath();
+  ctx.arc(this.x, this.y, this.arcR, this.startAngle, this.endAngle, this.counterClockWise);
```

Comment:
You are using `this.arcR` instead of `this.r` (from line 53)

Reply:
Thank you Andy